editor/effects.js

```diff
@@ -20,9 +20,30 @@ import {
 	getCurrentPostType,
 	getBlocks,
 	getPostEdits,
+	isCleanNewPost,
 } from './selectors';
 
+let originalDocumentTitle;
+
+function populateDocumentTitle( title, isCleanNew = false ) {
+	if ( ! title.trim() ) {
+		title = isCleanNew ? __( 'New post' ) : __( '(Untitled)' );
+	}
+	if ( ! originalDocumentTitle ) {
```

Comment:
Instead of tracking this as inferred from the original title, should not the client be able to generate the desired title at any point in time, including when the editor first initializes?